app/src/main/java/com/frinder/frinder/dataaccess/UserFirebaseDas.java

```diff
@@ -34,25 +36,55 @@ public UserFirebaseDas(Context context) {
         db = FirebaseFirestore.getInstance();
     }
 
-    public void addUser(User user){
-        Map<String, Object> usr = convertToFirebaseObject(user);
+    public void updateUserLocation(String id, ArrayList location) {
+        DocumentReference userRef = db.collection("users").document(id);
 
-        db.collection("users").document(user.getUid())
-                .set(usr)
+        userRef.update("location", location)
                 .addOnSuccessListener(new OnSuccessListener<Void>() {
                     @Override
                     public void onSuccess(Void aVoid) {
-                        Log.d(TAG, "DocumentSnapshot successfully written!");
+                        Log.d(TAG, "DocumentSnapshot successfully updated!");
                     }
                 })
                 .addOnFailureListener(new OnFailureListener() {
                     @Override
                     public void onFailure(@NonNull Exception e) {
-                        Log.w(TAG, "Error writing document", e);
+                        Log.e(TAG, "Error updating document", e);
                     }
                 });
     }
 
+    public void addUser(final User user){
+        LocationUtils locationUtils = new LocationUtils(context);
+        locationUtils.startLocationUpdates();
+        locationUtils.getLastLocation(new LocationUtils.LocationUpdate() {
+            @Override
+            public void onSuccess(Location location) {
```

Comment:
We do not call onSuccess in some cases (e.g. permission was not present). In that case, we would not add the user to firebase. This might break the app/server, where we have a profile that is not on the server.